src/services/queue/sqs-queue-service.ts

```diff
@@ -105,10 +126,37 @@ export class SqsQueueService<TInput, TValue> implements IQueueService<TValue> {
       output.Messages[0] as Message
     )
   }
+
+  /**
+   * Publishes a message to a sqs queue
+   * @param data the data you want to publish
+   */
+  async sendMessage(data: TValue): Promise<void> {
+    const sendMessageCommandInput = {
+      QueueUrl: this.sqsQueueUrl,
+      MessageBody: JSON.stringify(this.messageType.encode(data)),
+    }
+
+    await this.sqsClient.send(new SendMessageCommand(sendMessageCommandInput))
+  }
+}
+
+/**
+ * AnchorBatchSqsQueueService is used to consume and publish anchor batch messages. These batches are anchored by anchor workers
+ */
+export class AnchorBatchSqsQueueService extends SqsQueueService<AnchorBatch> {
+  constructor(config: Config) {
+    super(config, config.queue.sqsBatchQueueUrl, AnchorBatch)
+  }
 }
 
-export class AnchorBatchSqsQueueService<TInput> extends SqsQueueService<TInput, AnchorBatch> {
+/**
+ * FailureSqsQueueService is used to consume and publish any failures that could happen during anchoring by the anchor workers.
+ * ex. A batch was partially completed so we can publish the partially complete batch to the failure queue.
+ * The failure queue can then handle it accordingly (such as putting the failed requests into a new batch) and provide alerts.
+ */
+export class FailureSqsQueueService extends SqsQueueService<QueueMessageData> {
```

Comment:
We're not using this yet, right? Just the queue implementation?

Reply:
yup exactly. I thought about it but nack is working sufficiently until we get into more complex failure handing situations